### ISSUE TITLE
Navigate to upper/lower categories

### ISSUE THREAD

**Issue by nicolas-raoul** [nicolas-raoul/apps-android-commons#66]
When looking for a category, it is hard to get the right level.

For instance let's say I want to find `Office buildings in Manhattan, New York City` but my search only gets `Buildings in New York by city` and `High-rise office buildings in Midtown Manhattan`. One is too broad and one is wrong (my pic is not in Midtown).

In such a case, it would be incredibly useful if I could, for each category proposed:
- Drill down, to reveal sub-categories
- Drill up, to reveal categories of this category

In the UI, it could be a "zoom" button on the left that launch a drill down/up popup.

![issue](https://cloud.githubusercontent.com/assets/99590/13080777/878a6532-d50d-11e5-8b80-8b116ab8d035.png)

**Comment by nicolas-raoul** [comment]
A kind of "close" button would also be needed to go back to the list of suggestions.

**Comment by nicolas-raoul** [comment]
When (and if) Structured Commons will become a reality, there will most certainly be a kind of replacement for categories, for instance there will be a way to say that my picture is a picture of a dog. The same UI as above will be needed, to allow me to select a particular subclass of dog. On the opposite, if searching for "airport sign" only makes "railway sign" and other vaguely related classes appear as results, I will be happy to be able to go to the parent class of "railway sign" to get the best existing class for the thing depicted in my picture.

**Comment by nicolas-raoul** [comment]
Two years later, the app now allows to explore categories (media, subcategories, parent categories):
![42507578-438d001c-8463-11e8-9874-4e1ad35dc19f](https://user-images.githubusercontent.com/99590/42554792-2fe9d274-8521-11e8-9531-9b46ab8f2f9b.png)
It sounds that this "Explore" feature could be reused to improve categorization, with the benefits described by this issue.

**Comment by ujjwalagrawal17** [comment]
Really nice idea. Maybe we will have to think of the UI as there can be multiple parent category so we can't just navigate the user to a particular one. We need to show user list of parent categories along with subcategories in 2tabs that too in a bottom sheet layout can be good.

**Comment by nicolas-raoul** [comment]
@ujjwalagrawal17 You are totally right that a category can have several parent categories. I was not aware of that two years ago. You are also right that #1244 can be solved at the same time.
In terms of UI, how about this:
![commons-issue66-mockup svg](https://user-images.githubusercontent.com/99590/42613414-c5edd576-85db-11e8-8ec9-5e954d631697.png)

- Screen A: A loop icon is visible at the left of each category. Pressing it leads to screen B
- Screen B: Shows the category's media, as an example to decide whether it is really the category you wanted (especially useful for non-native speakers as category names are only in English). Pressing the tabs lead to both screens C and D
- Screen C: subcategories. If I search for "apple" I will be able to see what subcategories of apple are available and decide which one is the most appropriate
- Screen D: parent categories. Allows you to see whether a slightly different category actually represents better what you meant.
- You can navigate up and down as long as you want, until you have selected the categories you want. 
 Then press the "Done" button (there might be better ways to implement this? any ideas?) These categories then appear as selected on screen A.

**Comment by ujjwalagrawal17** [comment]
We can maybe put the done text in the menu (toolbar).

**Comment by nicolas-raoul** [comment]
Yes good idea! :-)

**Comment by ujjwalagrawal17** [comment]
Can we place the select this category icon in the menu too ? Tick icon and DONE TEXT on the right .

**Comment by nicolas-raoul** [comment]
Do like you want, just make sure it is visible at all times 🙂

**Comment by neslihanturan** [comment]
It seems like we already made some progress for this enhancement. If you are a beginner and you have read this thread until here, I want to sum up this issue for you.

We already have sub, parent categories logic which is in use at our explore activity. Now you can use that logic to browse sub/parent categories of suggested categories. The screenshot I shared can clarify what I mean. I think this task can be proper for a beginner.
![image](https://user-images.githubusercontent.com/3127881/54115568-be4cc580-43fd-11e9-8343-7972d59a020e.png)

**Comment by shalishi** [comment]
Hi, I would like take this issue. Can I?

**Comment by nicolas-raoul** [comment]
@shalishi OK, it is yours :-)
Please let us know about your progress every few days, thanks!

**Comment by Rishavgupta12345** [comment]
This issue is pretty old and irrelevant now. The app has gone through many changes and this is no longer an issue. please verify and close this issue @nicolas-raoul

**Comment by nicolas-raoul** [comment]
@Rishavgupta12345  Still relevant, still needed.

**Comment by nicolas-raoul** [comment]
@Rishavgupta12345 Thanks! :-)

**Comment by Rishavgupta12345** [comment]
@nicolas-raoul  I am unable to find the activity/fragment which has the scrren A in https://github.com/commons-app/apps-android-commons/issues/66#issuecomment-404389721 , please help me

**Comment by nicolas-raoul** [comment]
@Rishavgupta12345 It is the category selection activity, it looks like this now:
![adb](https://user-images.githubusercontent.com/99590/152104937-50991ec0-2104-49dc-8c34-81487e13ec50.png)

**Comment by Rishavgupta12345** [comment]
> @Rishavgupta12345 It is the category selection activity, it looks like this now: ![adb](https://user-images.githubusercontent.com/99590/152104937-50991ec0-2104-49dc-8c34-81487e13ec50.png) You could put the search loop on the right.

I think you are referring to the wrong category selection, the one you are referring is present while uploading a new photo, BUT the one this issue is talking about is:- when you go to explore option and try to search a picture , check the issue from beginning

**Comment by nicolas-raoul** [comment]
@Rishavgupta12345 This issue is about being able to find and select parent and children categories **when uploading** a picture. I updated the title to make this clearer.

**Comment by Rishavgupta12345** [comment]
> @Rishavgupta12345 This issue is about being able to find and select parent and children categories **when uploading** a picture. I updated the title to make this clearer.

this isn't a good first issue any more SO I think i should not work on it

**Comment by nicolas-raoul** [comment]
@Rishavgupta12345  I indeed think it is a bit hard for a first issue. But since it was marked as a good first issue when you started working on it, we let you choose whether you want to work on it or not. :-)

**Comment by nicolas-raoul** [comment]
Today I learnt that on the website, [HotCat](https://en.wikipedia.org/wiki/Wikipedia:HotCat) can do this.
It is hard to find andcan be tricky (in some cases it seems that I have to add a space at the end of my search to make suggestions appear) so it is not necessarily an example to follow, but could give us some insights.
<img width="321" alt="Screen Shot 2022-02-13 at 22 56 11" src="https://user-images.githubusercontent.com/99590/153756557-98ba97d9-c785-4ab4-902c-71687ec3ce8d.png">
